File: Ejercicios_Varios/Java/Proyecto_Java/src/main/java/Ejecicios/Ejercicio_3.java
package Ejecicios;

import java.util.Scanner;

public class Ejercicio_3 {

    /*
    Construir un programa que, al recibir como datos N numeros naturales
    determine cuantos de ellos son positivos, negativos o nulos.

    Si el usuario escribe un numero incorrecto, el programa no se ejecuta.
    En cambio, pregunta de nuevo por la informacion hasta que el dato ingresado sea corrrecto.
     */

    public static void main(String[] args) {

        while (true) {

            Scanner input = new Scanner(System.in);
            System.out.println("ingrese la cantidad de datos a analizar: ");
            int numero = input.nextInt();
            int positivo = 0, negativo = 0, cero = 0;

            if (numero > 0) {

                for (int i = 0; i < numero; i++) {
                    System.out.println("Ingrese un numero natural: ");

                    int numero_nat = input.nextInt();

                    if (numero_nat > 0) {

                        positivo++;
                    } else if (numero_nat < 0) { //else if traducuria "por otro lado si.."
                        //System.out.println("Numero negativo: ");
                        negativo++;
                    } else {
                        cero++;
                    }

                }


                System.out.println("la cantidad de numeros positvos es: " + positivo);
                System.out.println("la cantidad de numeros negativos es: " + negativo);
                System.out.println("la cantidad de numeros nulos es: " +     cero);

                {
                    break;
                }


            } else {
                System.out.println("El numero que ingreso no es correcto, vuelva a ingresar un numero");
            }

        }


    }

}

File: Ejercicios_Varios_2/Python/CursoPython/Ejercicios/Ejercicio3.py
'''
    Escribir un programa  que al escribir como dato un numero
    entero positivo N, Calcule el resultado de las siguientes series:
    1 + (1/2) + (1/3) +(1/4) ... (1/N)
    Si el usuario escribe un numero incorrecto, el programa no se ejecuta
    En cambio, pregunta de nuevo spor la informacion hasta que el dato sea correcto
'''
while (True):

        numero = int(input(f'ingrese un numero entero positivo: '))
        resultado = float(0)

        if numero > 0:
            print(f'el numero ingresado es correcto')
            for i in range(numero):

                resultado += (1 / float(i))
                print(f'el resultado es: {resultado}')
                print(type(resultado))
                "break"
        else:
            print(f'el numero ({numero}) no es entero positivo, vuelva a intentarlo')
